fdbclient/SpecialKeySpace.actor.cpp

```diff
@@ -1640,34 +1641,56 @@ ACTOR static Future<Optional<std::string>> coordinatorsCommitActor(ReadYourWrite
 		auto processesStr = processes_entry.second.get().toString();
-		boost::split(process_address_strs, processesStr, [](char c) { return c == ','; });
-		if (!process_address_strs.size()) {
+		boost::split(process_address_or_hostname_strs, processesStr, [](char c) { return c == ','; });
+		if (!process_address_or_hostname_strs.size()) {
 			return ManagementAPIError::toJsonString(
 			    false,
 			    "coordinators",
 			    "New coordinators\' processes are empty, please specify new processes\' network addresses with format "
-			    "\"IP:PORT,IP:PORT,...,IP:PORT\"");
+			    "\"IP:PORT,IP:PORT,...,IP:PORT\" or \"HOSTNAME:PORT,HOSTNAME:PORT,...,HOSTNAME:PORT\"");
 		}
-		for (index = 0; index < process_address_strs.size(); index++) {
+		for (index = 0; index < process_address_or_hostname_strs.size(); index++) {
 			try {
-				auto a = NetworkAddress::parse(process_address_strs[index]);
-				if (!a.isValid())
-					parse_error = true;
-				else
-					addressesVec.push_back(a);
+				if (Hostname::isHostname(process_address_or_hostname_strs[index])) {
+					hostnamesVec.push_back(Hostname::parse(process_address_or_hostname_strs[index]));
+				} else {
+					NetworkAddress a = NetworkAddress::parse(process_address_or_hostname_strs[index]);
+					if (!a.isValid()) {
+						parse_error = true;
+					} else {
+						addressesVec.push_back(a);
+					}
+				}
 			} catch (Error& e) {
 				TraceEvent(SevDebug, "SpecialKeysNetworkParseError").error(e);
 				parse_error = true;
 			}
 
 			if (parse_error) {
-				std::string error =
-				    "ERROR: \'" + process_address_strs[index] + "\' is not a valid network endpoint address\n";
-				if (process_address_strs[index].find(":tls") != std::string::npos)
+				std::string error = "ERROR: \'" + process_address_or_hostname_strs[index] +
+				                    "\' is not a valid network endpoint address\n";
+				if (process_address_or_hostname_strs[index].find(":tls") != std::string::npos)
 					error += "        Do not include the `:tls' suffix when naming a process\n";
 				return ManagementAPIError::toJsonString(false, "coordinators", error);
 			}
 		}
+
+		std::vector<Future<Void>> fs;
+		for (const auto& hostname : hostnamesVec) {
+			fs.push_back(map(INetworkConnections::net()->resolveTCPEndpoint(hostname.host, hostname.service),
+			                 [=](std::vector<NetworkAddress> const& addresses) -> Void {
+				                 NetworkAddress address =
+				                     addresses[deterministicRandom()->randomInt(0, addresses.size())];
+				                 address.flags = 0; // Reset the parsed address to public
+				                 address.fromHostname = NetworkAddressFromHostname::True;
+				                 if (hostname.isTLS) {
+					                 address.flags |= NetworkAddress::FLAG_TLS;
+				                 }
+				                 addressesVec.push_back(address);
+				                 return Void();
+			                 }));
+		}
+		wait(waitForAll(fs));
```

Comment:
Do we actually need to do this? My understanding of coordinators command is to update the connection string. The server side can already handle connection string with hostnames. So won't we just need to update the connection (hostname version) in the system?